sktime/alignment/base.py

```diff
@@ -336,4 +336,12 @@ def _get_distance_matrix(self):
         distmat: an (n x n) np.array of floats, where n is length of X passed to fit
             [i,j]-th entry is alignment distance between X[i] and X[j] passed to fit
         """
-        raise NotImplementedError
+        import numpy as np
```

Comment:
Should this import statement be at the beginning of the file since numpy is no soft-dependency?

Reply:
hm, I thought, as it is only needed when the default is not overridden, it minimizes coupling if it is here.